tool/release/CreateNotes.kt

```diff
@@ -0,0 +1,216 @@
+            .setHeaders(
+                HttpHeaders().setAuthorization("Token $githubToken").setAccept("application/vnd.github.v3+json")
+            )
+            .execute()
+        val body = Json.parse(String(response.content.readBytes()))
+        val prs = body.asArray()
+        if (prs.size() > 0) {
+            val notes = prs.map { pr ->
+                val types = pr.asObject().get("labels").asArray().map { e -> e.asObject().get("name").asString() }
+                    .filter { e -> e.startsWith("type:") }
+                val type =
+                    if (types.contains("type: feature")) CommitNoteType.FEATURE
+                    else if (types.contains("type: bug")) CommitNoteType.BUG
+                    else if (types.contains("type: refactor")) CommitNoteType.REFACTOR
+                    else CommitNoteType.OTHER
+                CommitNote(
+                    title = pr.asObject().get("title").asString(),
+                    desc = pr.asObject().get("body").asString(),
+                    type = type
+                )
+            }
+            notes
+        } else {
+            val response = NetHttpTransport()
+                .createRequestFactory()
+                .buildGetRequest(GenericUrl("https://api.github.com/repos/$org/$repo/commits/$commit"))
+                .setHeaders(
+                    HttpHeaders().setAuthorization("Token $githubToken").setAccept("application/vnd.github.v3+json")
+                )
+                .execute()
+            val body = Json.parse(String(response.content.readBytes()))
+            val notes = listOf(
+                CommitNote(
+                    title = body.asObject().get("commit").asObject().get("message").asString(),
+                    desc = "",
+                    type = CommitNoteType.OTHER
+                )
+            )
+            notes
+        }
+    }
+}
+
+fun createCommitNote(note: CommitNote): String {
+    var desc = StringBuilder()
+    var header = 0
+    for (line in note.desc.lines()) {
+        if (line.startsWith("##")) {
+            header += 1
+        } else if (header == 1) {
+            desc.append(line)
+        } else if (header > 1) {
+            break
+        }
+    }
+    return "- **${note.title}**\n  $desc"
+}
+
+fun createReleaseNote(commitNotes: List<CommitNote>, releaseTemplateFile: Path) {
+    val features = mutableListOf<CommitNote>()
+    val bugs = mutableListOf<CommitNote>()
+    val refactors = mutableListOf<CommitNote>()
+    val others = mutableListOf<CommitNote>()
+
+    commitNotes.forEach { note ->
+        when (note.type) {
+            CommitNoteType.FEATURE -> features.add(note)
+            CommitNoteType.BUG -> bugs.add(note)
+            CommitNoteType.REFACTOR -> refactors.add(note)
+            else -> others.add(note)
+        }
+    }
+
+    val releaseNote = """
+Install & Run: http://docs.vaticle.com/docs/running-typedb/install-and-run
+
+## New Features
```

Comment:
@lolski code below could be generified with something like

```
val compileNotes = { notes: Iterable<CommitNote> -> notes.map(::createCommitNote).joinToString("\n") }
```

and then in the string e.g.:

```
## Bugs Fixed
${compileNotes(bugs)}
```

Reply:
Hmm I think we should keep the current code, because your suggestion is too terse.